storage: refactor consistency check runner

tbg opened the pull request:
In preparation for checking stats during consistency checker runs,
separate out the phases of obtaining results and acting upon them.

Special care was taken to make the code more symmetrical in minimizing
the role of the local replica. This makes the code more transparent.

Release note: None

tbg committed:
storage: refactor consistency check runner

In preparation for checking stats during consistency checker runs,
separate out the phases of obtaining results and acting upon them.

Special care was taken to make the code more symmetrical in minimizing
the role of the local replica. This makes the code more transparent.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21131)
<!-- Reviewable:end -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21131#-:-L1ZX4X6dY6U10cXPaCd:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tbg committed:
storage: provide resolver in DummyRaftTransport

Providing a nil resolver provokes NPEs and requires awkward workarounds
and awareness that a resolver can be nil at all times (in certain tests).

The previous commit would fail quickly under

`make stressrace PKG=./pkg/storage/ TESTS=TestStoreZoneUpdateAndRangeSplit`

since the consistency checker would run and try to use the resolver.

Release note: None

tschottdorf commented:
TFTR! I added another commit that addresses NPEs during tests, but it's unlikely to be controversial so I'll merge on green.
